src/libs.rs

```diff
@@ -0,0 +1,104 @@
+use std::env;
+use std::error::Error;
+use std::fs;
+
+pub struct Config {
+    pub query: String,
+    pub file_path: String,
+    pub not_case_sensitive: bool,
+}
+
+impl Config {
+    pub fn build(args: &Vec<String>) -> Result<Config, &'static str> {
+        if args.len() < 3 {
+            return Err("Not enough arguments");
+        }
+
+        let query = args[1].clone();
+        let file_path = args[2].clone();
+        let not_case_sensitive =
+            env::var("NOT_CASE_SENSITIVE").is_ok() || args.contains(&String::from("-cs"));
+
+        Ok(Config {
+            query,
+            file_path,
+            not_case_sensitive,
+        })
+    }
+}
+
+pub fn read_from_file(file_path: &String) -> Result<String, Box<dyn Error>> {
+    let contents = fs::read_to_string(file_path).expect("Something went wrong reading the file");
+    Ok(contents)
+}
+
+pub fn get_lines_with_query(
+    file_content: &String,
+    query: &String,
+    not_case_sensitive: &bool,
+) -> Vec<String> {
+    let mut results: Vec<String> = Vec::new();
+
+    for line in file_content.lines() {
+        if (!not_case_sensitive && line.contains(query))
+            || (*not_case_sensitive && line.to_lowercase().contains(&query.to_lowercase()))
```

Comment:
Can we compute query/line to lowercase before so that both `line` and `query` are what they need to be already at that point? It can make code look better.